src/lib/comms/include/sol-gatt.h

```diff
@@ -0,0 +1,256 @@
+/*
+ * This file is part of the Soletta Project
+ *
+ * Copyright (C) 2016 Intel Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <sol-bluetooth.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @file
+ * @brief Routines to handle Bluetooth GATT protocol
+ *
+ * The Bluetooth Generic Attribute Protocol (GATT) is ...
+ */
+
+enum sol_gatt_chr_flags {
+	SOL_GATT_CHR_FLAGS_BROADCAST = (1 << 0),
+	SOL_GATT_CHR_FLAGS_READ = (1 << 1),
+	SOL_GATT_CHR_FLAGS_WRITE_WITHOUT_RESPONSE = (1 << 2),
+	SOL_GATT_CHR_FLAGS_WRITE = (1 << 3),
+	SOL_GATT_CHR_FLAGS_NOTIFY = (1 << 4),
+	SOL_GATT_CHR_FLAGS_INDICATE = (1 << 5),
+	SOL_GATT_CHR_FLAGS_AUTHENTICATED_SIGNED_WRITES = (1 << 6),
+	SOL_GATT_CHR_FLAGS_RELIABLE_WRITE = (1 << 7),
+	SOL_GATT_CHR_FLAGS_WRITABLE_AUXILIARIES = (1 << 8),
+	SOL_GATT_CHR_FLAGS_ENCRYPT_READ = (1 << 9),
+	SOL_GATT_CHR_FLAGS_ENCRYPT_WRITE = (1 << 10),
+	SOL_GATT_CHR_FLAGS_ENCRYPT_AUTHENTICATED_READ = (1 << 11),
+	SOL_GATT_CHR_FLAGS_ENCRYPT_AUTHENTICATED_WRITE = (1 << 12),
+};
+
+enum sol_gatt_desc_flags {
+	SOL_GATT_CHR_FLAGS_READ = (1 << 0),
+	SOL_GATT_CHR_FLAGS_WRITE = (1 << 1),
+	SOL_GATT_CHR_FLAGS_ENCRYPT_WRITE = (1 << 2),
+	SOL_GATT_CHR_FLAGS_ENCRYPT_AUTHENTICATED_READ = (1 << 3),
+	SOL_GATT_CHR_FLAGS_ENCRYPT_AUTHENTICATED_WRITE = (1 << 4),
+};
+
+/*
+ * FIXME: need to add errors that make sense for the BlueZ D-Bus API
+ * and "raw" GATT.
+ */
+enum sol_gatt_error {
```

Comment:
usually we're not using specific errors like this, rather translating them to errno